feature-libs/cart/base/components/clear-cart/clear-cart-button/clear-cart.component.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  ViewChild,
+  OnInit,
+  ViewContainerRef,
+} from '@angular/core';
+import { LaunchDialogService, LAUNCH_CALLER } from '@spartacus/storefront';
+import { ActiveCartFacade, Cart } from '@spartacus/cart/base/root';
+import { Observable, Subscription } from 'rxjs';
+import { take } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-clear-cart',
+  templateUrl: './clear-cart.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ClearCartComponent implements OnInit {
+  cart$: Observable<Cart>;
+  private subscription = new Subscription();
+  @ViewChild('element') element: ElementRef;
+
+  constructor(
+    protected activeCartService: ActiveCartFacade,
+    protected vcr: ViewContainerRef,
+    protected launchDialogService: LaunchDialogService
+  ) {}
+
+  ngOnInit(): void {
+    this.cart$ = this.activeCartService.getActive();
+  }
```

Comment:
We tend to try and initialize class attribute where they are declared instead.  In this case we could also then remove   ngOnInit()